--- /dev/fd/63	2019-11-28 17:43:22.000000000 -0800
+++ ../R Markdown.sublime-syntax	2019-11-28 17:36:56.000000000 -0800
@@ -1,13 +1,13 @@
 %YAML 1.2
 ---
 # http://www.sublimetext.com/docs/3/syntax.html
-name: Markdown
+name: R Markdown
 file_extensions:
-  - md
-  - mdown
-  - markdown
-  - markdn
-scope: text.html.markdown
+  - rmd
+  - rmdown
+  - rmarkdown
+  - rmarkdn
+scope: text.html.markdown.rmarkdown
 comment: |-
   this definition aims to meet CommonMark specifications http://spec.commonmark.org/
   with GitHub Formatted Markdown extensions              https://github.github.com/gfm/
@@ -181,6 +181,31 @@
     #   )
 contexts:
   main:
+    - match: (?=^\s*(?:---|\+\+\+)\s*$)
+      push: front-matter
+    - match: ""
+      set: body
+  front-matter:
+    - match: ^\s*(---)\s*$
+      captures:
+        1: punctuation.definition.frontmatter
+      embed: scope:source.yaml
+      embed_scope: markup.raw.yaml.front-matter
+      escape: ^\s*(---)\s*$
+      escape_captures:
+          1: punctuation.definition.frontmatter
+    - match: ^\s*(\+\+\+)\s*$
+      captures:
+        1: punctuation.definition.frontmatter
+      embed: scope:source.toml
+      embed_scope: markup.raw.toml.front-matter
+      escape: ^\s*(\+\+\+)\s*$
+      escape_captures:
+          1: punctuation.definition.frontmatter
+    - match: ""
+      pop: true
+      set: body
+  body:
     - match: |-
         (?x)^
         (?=  {{block_quote}}
@@ -458,6 +483,7 @@
               pop: true
             - include: inline
             - include: bold
+            - include: supersubscript
             - include: bold-italic-trailing
         - match: \*
           scope: punctuation.definition.italic.end.markdown
@@ -475,6 +501,7 @@
               pop: true
             - include: inline
             - include: italic
+            - include: supersubscript
             - include: bold-italic-trailing
         - include: inline
         - include: bold-italic-trailing
@@ -495,6 +522,7 @@
         - match: \b_(?=[^\s_])(?=[^*_]*\*\*)
           comment: eat the underscore that has no corresponding underscore before the closing bold punctuation on the same line, as it won't be treated as italic by CommonMark
         - include: italic
+        - include: supersubscript
         - include: bold-italic-trailing
     - match: '\b(__)(_)(?=\S)(?!_)'
       captures:
@@ -527,6 +555,7 @@
               pop: true
             - include: inline
             - include: italic
+            - include: supersubscript
             - include: bold-italic-trailing
         - match: __\b
           scope: punctuation.definition.bold.end.markdown
@@ -542,6 +571,7 @@
               pop: true
             - include: inline
             - include: bold
+            - include: supersubscript
             - include: bold-italic-trailing
         - include: inline
         - include: bold-italic-trailing
@@ -562,6 +592,7 @@
         - match: \*(?=[^\s*])(?=[^*_]*__\b)
           comment: eat the asterisk that has no corresponding asterisk before the closing bold punctuation on the same line, as it won't be treated as italic by CommonMark
         - include: italic
+        - include: supersubscript
         - include: bold-italic-trailing
   bracket:
     - match: '<(?![A-Za-z/?!$])'
@@ -705,6 +736,7 @@
     - include: ampersand
     - include: ligatures
     - include: bracket
+    - include: latex-math
     - include: raw
     - include: image-inline
     - include: link-inline
@@ -714,11 +746,13 @@
     - include: link-ref-literal
     - include: link-ref
     - include: link-ref-footnote
+    - include: citations
   inline-bold-italic:
     - include: inline
     - include: bold
     - include: italic
     - include: strikethrough
+    - include: supersubscript
   inline-bold-italic-linebreak:
     - include: inline-bold-italic
     - include: hard-line-break
@@ -739,6 +773,7 @@
           pop: true
         - include: inline
         - include: bold
+        - include: supersubscript
         - match: '\*+'
         - include: bold-italic-trailing
     - match: '\b_(?=\S)(?!_)'
@@ -757,19 +792,44 @@
           pop: true
         - include: inline
         - include: bold
+        - include: supersubscript
         - include: bold-italic-trailing
     - match: '[*_]+'
+  supersubscript:
+    - match: '(~)(?=([^~ \n]|\\ )+~)'
+      scope: punctuation.definition.subscript.begin.markdown
+      push:
+        - meta_scope: markup.subscript.markdown-pandoc
+        - match: '~'
+          scope: punctuation.definition.subscript.end.markdown
+          pop: true
+        - include: inline
+        - include: bold
+        - include: italic
+        - include: bold-italic-trailing
+    - match: '(\^)(?=([^\^ \n]|\\ )+\^)'
+      scope: punctuation.definition.superscript.begin.markdown
+      push:
+        - meta_scope: markup.superscript.markdown-pandoc
+        - match: \^
+          scope: punctuation.definition.superscript.end.markdown
+          pop: true
+        - include: inline
+        - include: bold
+        - include: italic
+        - include: bold-italic-trailing
   strikethrough:
-    - match: '(~+)(?=\S)(?!~)'
+    - match: '(~~+)(?=\S)(?!~)'
       scope: punctuation.definition.strikethrough.begin.markdown
       push:
         - meta_scope: markup.strikethrough.markdown-gfm
-        - match: ~+
+        - match: ~~+
           scope: punctuation.definition.strikethrough.end.markdown
           pop: true
         - include: inline
         - include: bold
         - include: italic
+        - include: supersubscript
         - include: bold-italic-trailing
   bold-italic-trailing:
     - include: scope:text.html.basic
@@ -967,6 +1027,12 @@
         1: punctuation.definition.link.begin.markdown
         2: meta.link.reference.literal.footnote-id.markdown
         3: punctuation.definition.link.end.markdown
+  citations:
+    - match: '(@)([\w:-]+)'
+      scope: meta.link.reference.markdown-pandoc
+      captures:
+        1: constant.other.cite.markdown-pandoc
+        2: keyword.other.cite.markdown-pandoc
   list-paragraph:
     - match: '^(?=(?:[ ]{4}|\t){2,}(?![>+*\s-]))(?={{indented_code_block}})'
       push:
@@ -1362,7 +1428,9 @@
       push:
         - match: $
           pop: true
+        - include: fenced-code-block-knitr
         - include: fenced-code-block
+    - include: code-span-knitr
     - include: code-span
   thematic-break:
     - match: '(?={{thematic_break}})'
@@ -1419,6 +1487,7 @@
           pop: true
     - include: bold
     - include: italic
+    - include: supersubscript
     - include: hard-line-break
     - include: scope:text.html.basic
   link-text-allow-image:
@@ -1560,6 +1629,7 @@
                       push:
                         - include: bold
                         - include: italic
+                        - include: supersubscript
                         - match: ''
                           pop: true
                     - match: |-
@@ -1570,6 +1640,198 @@
                     - include: inline
                     - include: scope:text.html.basic
 
+  fenced-code-block-knitr:
+    - match: (```)(\{)(r|Rscript)\b
+      captures:
+        1: punctuation.definition.raw.code-fence.begin.markdown
+        2: punctuation.definition.raw.code-fence.options.begin.markdown
+        3: constant.other.language-name.markdown
+      push:
+        - meta_scope: meta.code-fence.definition.begin.r.markdown-knitr
+        - meta_content_scope: meta.code-fence.definition.optio.markdown-knitr
+        - match: (?=\})
+          set:
+            - match: '\}'
+              scope: meta.code-fence.definition.begin.r.markdown-knitr punctuation.definition.raw.code-fence.options.end.markdown
+              embed: scope:source.r
+              embed_scope: markup.raw.code-fence.r.markdown-knitr
+              escape: '(```)'
+              escape_captures:
+                0: meta.code-fence.definition.end.r.markdown-knitr
+                1: punctuation.definition.raw.code-fence.end.markdown
+            - match: ''
+              pop: true
+        - include: scope:source.r
+
+    - match: (```)(\{)(python)\b
+      captures:
+        1: punctuation.definition.raw.code-fence.begin.markdown
+        2: punctuation.definition.raw.code-fence.options.begin.markdown
+        3: constant.other.language-name.markdown
+      push:
+        - meta_scope: meta.code-fence.definition.begin.python.markdown-knitr
+        - meta_content_scope: meta.code-fence.definition.options.py.markdown-knitr
+        - match: (?=\})
+          set:
+            - match: '\}'
+              scope: meta.code-fence.definition.begin.python.markdown-knitr punctuation.definition.raw.code-fence.options.end.markdown
+              embed: scope:source.python
+              embed_scope: markup.raw.code-fence.python.markdown-knitr
+              escape: '(```)'
+              escape_captures:
+                0: meta.code-fence.definition.end.python.markdown-knitr
+                1: punctuation.definition.raw.code-fence.end.markdown
+            - match: ''
+              pop: true
+        - include: scope:source.r
+
+    - match: (```)(\{)(julia)\b
+      captures:
+        1: punctuation.definition.raw.code-fence.begin.markdown
+        2: punctuation.definition.raw.code-fence.options.begin.markdown
+        3: constant.other.language-name.markdown
+      push:
+        - meta_scope: meta.code-fence.definition.begin.julia.markdown-knitr
+        - meta_content_scope: meta.code-fence.definition.options.j.markdown-knitr
+        - match: (?=\})
+          set:
+            - match: '\}'
+              scope: meta.code-fence.definition.begin.julia.markdown-knitr punctuation.definition.raw.code-fence.options.end.markdown
+              embed: scope:source.julia
+              embed_scope: markup.raw.code-fence.julia.markdown-knitr
+              escape: '(```)'
+              escape_captures:
+                0: meta.code-fence.definition.end.julia.markdown-knitr
+                1: punctuation.definition.raw.code-fence.end.markdown
+            - match: ''
+              pop: true
+        - include: scope:source.r
+
+    - match: (```)(\{)(sql)\b
+      captures:
+        1: punctuation.definition.raw.code-fence.begin.markdown
+        2: punctuation.definition.raw.code-fence.options.begin.markdown
+        3: constant.other.language-name.markdown
+      push:
+        - meta_scope: meta.code-fence.definition.begin.sql.markdown-knitr
+        - meta_content_scope: meta.code-fence.definition.options.markdown-knitr
+        - match: (?=\})
+          set:
+            - match: '\}'
+              scope: meta.code-fence.definition.begin.sql.markdown-knitr punctuation.definition.raw.code-fence.options.end.markdown
+              embed: scope:source.sql
+              embed_scope: markup.raw.code-fence.sql.markdown-knitr
+              escape: '(```)'
+              escape_captures:
+                0: meta.code-fence.definition.end.sql.markdown-knitr
+                1: punctuation.definition.raw.code-fence.end.markdown
+            - match: ''
+              pop: true
+        - include: scope:source.r
+
+    - match: (```)(\{)(bash)\b
+      captures:
+        1: punctuation.definition.raw.code-fence.begin.markdown
+        2: punctuation.definition.raw.code-fence.options.begin.markdown
+        3: constant.other.language-name.markdown
+      push:
+        - meta_scope: meta.code-fence.definition.begin.bash.markdown-knitr
+        - meta_content_scope: meta.code-fence.definition.options..markdown-knitr
+        - match: (?=\})
+          set:
+            - match: '\}'
+              scope: meta.code-fence.definition.begin.bash.markdown-knitr punctuation.definition.raw.code-fence.options.end.markdown
+              embed: scope:source.shell.bash
+              embed_scope: markup.raw.code-fence.bash.markdown-knitr
+              escape: '(```)'
+              escape_captures:
+                0: meta.code-fence.definition.end.bash.markdown-knitr
+                1: punctuation.definition.raw.code-fence.end.markdown
+            - match: ''
+              pop: true
+        - include: scope:source.r
+
+
+  code-span-knitr:
+    - match: (`+)(r) (?!`)
+      captures:
+        1: punctuation.definition.raw.begin.markdown
+        2: constant.other.language-name.markdown
+      push:
+        - meta_scope: markup.raw.inline.markdown-knitr
+        - meta_content_scope: source.r
+        - match: \1(?!`)
+          scope: punctuation.definition.raw.end.markdown
+          pop: true
+        - match: '`+'
+        - match: ^\s*$\n?
+          scope: invalid.illegal.non-terminated.raw.markdown
+          pop: true
+        - include: scope:source.r
+
+  latex-math:
+    - match: \\\$
+    - match: \$\$
+      scope: string.other.math.latex punctuation.definition.string.begin.latex
+      push:
+        - meta_scope: text.tex.latex meta.environment.math.latex
+        - include: scope:text.tex.latex#macros
+        - include: math-content
+        - match: \$\$
+          scope: string.other.math.latex punctuation.definition.string.end.latex
+          pop: true
+    - match: |-
+        (?x)
+        (\$)(?=\S)
+        (?=
+          (?:
+            \\\\
+            |\\\$
+            |[^\$]
+          )*?
+          \S\$(?:[^a-zA-Z0-9]|$)
+        )
+      scope: string.other.math.latex punctuation.definition.string.begin.latex
+      push:
+        - meta_scope: text.tex.latex meta.environment.math.latex
+        - include: scope:text.tex.latex#macros
+        - include: math-content
+        - match: \$
+          scope: string.other.math.latex punctuation.definition.string.end.latex
+          pop: true
+    - match: '(?=\\begin\{\s*(equation|align)\s*\*?\s*\})'
+      push:
+        - include: math-content
+        - match: '(?=\S)'
+          pop: true
+
+  math-braces:
+    - match: '\{'
+      scope: punctuation.definition.group.brace.begin.latex
+      push:
+        - meta_scope: meta.group.brace.latex
+        - match: '\}'
+          scope: punctuation.definition.group.brace.end.latex
+          pop: true
+        - include: math-content
+
+  math-content:
+    - include: code-span-knitr
+    - include: scope:text.tex.latex#verb
+    - include: scope:text.tex.latex#text-decorators
+    - include: scope:text.tex.latex#references
+    - include: scope:text.tex.latex#begin-end-commands
+    - include: scope:text.tex#greeks
+    - include: scope:text.tex#math-brackets
+    - include: math-braces
+    - include: scope:text.tex.latex#boxes
+    - include: scope:text.tex#math-commands
+    - include: scope:text.tex#math-operators
+    - include: scope:text.tex#math-characters
+    - include: scope:text.tex#math-numerics
+    - include: scope:text.tex.latex#general-constants
+
+
 ###[ PROTOTYPES ]#############################################################
 
   else-pop:
